### PR TITLE
Add GQL023: AotSchema constructor must call Configure or chain to another constructor

### PR DESCRIPTION
**Pull request by Shane32**
## Summary
Adds a new analyzer (GQL023) that warns when an `AotSchema` constructor doesn't properly initialize the schema by calling `Configure()` or chaining to another constructor.

## Changes
- **New Analyzer**: `AotSchemaConstructorAnalyzer` (GQL023)
  - Warns when constructors in classes deriving from `AotSchema` don't call `Configure()` or use `: this(...)` constructor chaining
  - Severity: Warning
  - Helps prevent runtime configuration issues

- **Tests**: 8 comprehensive test cases covering:
  - Constructors that call `Configure()` directly
  - Constructor chaining patterns
  - Multiple constructor scenarios
  - Inheritance from derived `AotSchema` classes
  - Edge cases (conditional calls, wrong method names, non-AotSchema classes)

- **Documentation**: Complete user-facing docs at `docs2/site/docs/analyzers/gql023.md`
  - Includes examples of violations and fixes
  - Explains when to suppress warnings

## Motivation
The `AotSchema.Configure()` method is critical for proper schema initialization. Without calling it, schemas may not be properly configured, leading to runtime errors. This analyzer ensures developers follow the correct initialization pattern.

## Testing
All 8 tests pass:
- ✅ Constructor calls Configure() - no diagnostic
- ✅ Constructor chains to another constructor - no diagnostic  
- ✅ Constructor doesn't call Configure() or chain - diagnostic
- ✅ Multiple constructors with one missing Configure() - diagnostic
- ✅ Derived from derived AotSchema missing Configure() - diagnostic
- ✅ Non-AotSchema class - no diagnostic (not applicable)
- ✅ Configure() called in conditional block - no diagnostic
- ✅ Calls wrong Configure method - diagnostic

### CODE REVIEW

File: src/GraphQL.Analyzers.Tests/AotSchemaConstructorAnalyzerTests.cs [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+                {
+                    // Missing Configure() call
+                }
+
+                protected override void Configure()
+                {
+                    base.Configure();
+                    // My configuration
+                }
+            }
```
Comment (gao-artur) [inline src/GraphQL.Analyzers.Tests/AotSchemaConstructorAnalyzerTests.cs:165]: This use case looks problematic. The `Configure` method will be called twice: once by the base constructor and once by the derived constructor (if the reported error is fixed). While this is a user's mistake and GraphQL.NET shouldn't be responsible for it, I wouldn't test for it.
I'd recommend adding the following test cases:
1. Abstract class constructors are not reported if they don't call `Configure`
2. Derived class constructors are not reported when the base constructor calls `Configure`

File: src/GraphQL.Analyzers/AotSchemaConstructorAnalyzer.cs [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    {
+        foreach (var invocation in body.DescendantNodes().OfType<InvocationExpressionSyntax>())
+        {
+            // Reject explicit base.Configure()
+            if (invocation.Expression is MemberAccessExpressionSyntax ma &&
+                ma.Expression is BaseExpressionSyntax &&
+                ma.Name.Identifier.Text == Constants.MethodNames.Configure)
+            {
+                continue;
+            }
```
Comment (gao-artur) [inline src/GraphQL.Analyzers/AotSchemaConstructorAnalyzer.cs:97]: Why `base.Configure()` is rejected?

Comment (Shane32) [reply src/GraphQL.Analyzers/AotSchemaConstructorAnalyzer.cs:97]: The source generator generates an overriding method called Configure. Calling base.Configure would have no effect as it contains no code. The only reason Configure exists in AotSchema is just for convenience: so (1) a user’s constructor that calls Configure won’t show an error before the source generator runs (2) no error if no attributes and the source generator doesn’t generate it.

File: src/GraphQL.Analyzers.Tests/AotSchemaConstructorAnalyzerTests.cs (hunk 1 above)
Comment (Shane32) [reply src/GraphQL.Analyzers.Tests/AotSchemaConstructorAnalyzerTests.cs:165]: Good catch!

Comment (Shane32) [reply src/GraphQL.Analyzers.Tests/AotSchemaConstructorAnalyzerTests.cs:165]: I added code to ignore abstract and further derived classes.  AI was trying to implement it as you described but the code was getting exceedingly complex.  Feel free to try it yourself, although I think it's quite an edge case so I'm not worried about adding it.  The critical issue for this analyzer, calling Configure when creating a constructor for an AOT-generated schema, is covered in this PR.

**Review by gao-artur — APPROVED** [review]
